### PR TITLE
Implement arduino serial communications

### PR DESCRIPTION
**Pull request by jssjr**
This is a working (I think) implementation of the serial communications protocol that bridges the Arduino to the RoboRIO. To use this, you should call `Arduino.initialize()` when the robot starts, then call `Arduino.update()` periodically when the robot is running.

There is one not so nice thing about this implementation. The sensor data itself is assigned to static variables on the `Robot` class and the values are updated from inside of `Arduino.update()` function. This means the two classes must have detailed knowledge of one another, which is a rather poor design choice. A better pattern might be to pass in an object that contains the sensors we would like to update as a parameter to the update function or as part of the Arduino initialization to avoid this tightly coupled relationship. However, the current implementation makes it easy to reference sensor data by using `Robot.heading` (for example) from anywhere in your Commands or Subsystems so I went with that.

The other change I made was to move the code updating the SmartDashboard into it's own function called `Robot#updateSmartDashboard()` so it can be called from all `Robot#*Periodic()` functions alongside the `Arduino#update()` function without having to copy paste these lines into multiple places.

/cc @donutpixel and @BearsAreAwsome for :eyes: since we can fall back to using this implementation if necessary.

### CODE REVIEW

File: src/org/usfirst/frc/team4533/robot/Robot.java [hunk 1]
```diff
@@ -55,9 +59,26 @@ public void robotInit() {
...
     	
 	
+    /**
+     * This function updates our smart dashboard values. It should probably be called periodically 
+     */
+    public void updateSmartDashboard() {
+        SmartDashboard.putBoolean("Gear", DriveSystem.hasGear());
+        SmartDashboard.putBoolean("Climber On?", ClimbSystem.isOn());
+        SmartDashboard.putNumber("Front Distance", DriveSystem.ultraSonic());
+        String messageOfTheDay = "don't do school stay in drugs";
```
Comment (abrightwell) [inline src/org/usfirst/frc/team4533/robot/Robot.java:75]: Can we please remove this MotD? I get the desire to be playful, but it serves no purpose at this point in time.

File: src/org/usfirst/frc/team4533/robot/utils/Arduino.java [hunk 2]
```diff
@@ -2,10 +2,97 @@
...
+	private static SerialPort serial;
+	private static boolean ready = false;
+	private static String oldBuffer = "";
+
+	private static final char syncChar = '^';
+	private static final char separatorChar = '~';
+
+	/* Initialize serial communications to the Ardunio */
+	public static void initialize() {
+		serial = new SerialPort(115200, SerialPort.Port.kUSB);
```
Comment (abrightwell) [inline src/org/usfirst/frc/team4533/robot/utils/Arduino.java:17]: Is there a constant that can be used for the baud rate?  If not, then no worries, but I thought there was a `SerialPort.BaudRate.115200` or something like that.

**Review by abrightwell — COMMENTED** [review]
Understanding this is meant for helping the students continue moving forward, I've made a few comments regardless.

File: src/org/usfirst/frc/team4533/robot/utils/Arduino.java [hunk 3]
```diff
@@ -2,10 +2,97 @@
...
+
+			// Process the data packet if we've located a sync byte
+			if (buffer.charAt(i) == '^') {
+				c = i + 1;
+
+				// Count the number of separator chars we see until we have 3 of
+				// them
+				while (c < buffer.length() && numFields != 3) {
+					if (buffer.charAt(c) == separatorChar) {
+						c++;
```
Comment (BearsAreAwsome) [inline src/org/usfirst/frc/team4533/robot/utils/Arduino.java:58]: This should increment numFields as well.

**Pull request merged**